packages/editor/src/store/selectors.js

```diff
@@ -485,18 +486,29 @@ export function isEditedPostEmpty( state ) {
 /**
  * Returns true if the post can be autosaved, or false otherwise.
  *
- * @param  {Object}  state Global application state.
+ * @param  {Object} state    Global application state.
+ * @param  {Object} autosave An autosave object.
  *
  * @return {boolean} Whether the post can be autosaved.
  */
-export function isEditedPostAutosaveable( state ) {
+export function isEditedPostAutosaveable( state, autosave ) {
+	if ( autosave === undefined ) {
```

Comment:
Why did you decide to pass the autosave into `isEditedPostAutosaveable` instead of always retrieving it here?

Reply:
Selectors should be functionally pure, and I could only achieve that by passing the autosave state in as an argument now that it no longer lives in the same state tree.

Reply:
Makes sense, thanks for clarifying!